dashboards/resources/queries/common.libsonnet

```diff
@@ -0,0 +1,90 @@
+// Shared query builders for all resource dashboards
+local maxBy = 'k8s_cluster_name, k8s_namespace_name, k8s_pod_name, k8s_container_name';
+local podMaxBy = 'k8s_cluster_name, k8s_namespace_name, k8s_pod_name';
+
+local withRate(expr, filters, useRate) =
+  if useRate then
+    'rate(%s{%s}[$__rate_interval])' % [expr, filters]
+  else
+    '%s{%s}' % [expr, filters];
+
+local sumExpr(inner, groupBy) =
+  if groupBy == '' then
+    'sum(\n  %s\n)' % inner
+  else
+    'sum by (%s) (\n  %s\n)' % [groupBy, inner];
+
+local maxByContainer(expr) =
+  'max by (%s) (\n    %s\n  )' % [maxBy, expr];
+
+// Active pod filter (Pending=1 or Running=2), normalized to 1
+// pod phases are 1=Pending, 2=Running, 3=Succeeded, 4=Failed, 5=Unknown
+// known issue here: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/36819
+local activePodFilter(phaseFilters) =
+  |||
+    * on (%(podMaxBy)s)
+    group_left() clamp_max(
+      max by (%(podMaxBy)s) (
+        (k8s_pod_phase{%(filters)s} == 1) or (k8s_pod_phase{%(filters)s} == 2)
+      ), 1
+    )
+  ||| % { podMaxBy: podMaxBy, filters: phaseFilters };
+
+{
+  metric(metric, filters, groupBy='')::
+    sumExpr(
+      maxByContainer('%s{%s}' % [metric, filters]),
+      groupBy
+    ),
+
+  rate(metric, filters, groupBy='')::
```

Comment:
The use of `groupBy` is confusing because it does not actually translate to `group by (...)`, maybe just `by` is a more appropriate argument name? The same applies everywhere `groupBy` is used, as far as I can see.